src/node.cc

```diff
@@ -3833,6 +4200,13 @@ static void ParseArgs(int* argc,
 
   // Copy remaining arguments.
   const unsigned int args_left = nargs - index;
+
+  if (is_env && args_left) {
+    fprintf(stderr, "%s: non-option '%s' is not allowed in NODE_OPTIONS\n",
```

Comment:
I'm not a fan of the `non-option` part, to be honest. I'm not that concerned about differentiating between known options that are not supported and things that are not valid options in the first place.